src/WebJobs.Extensions.OpenAI/Search/SemanticSearchConverter.cs

```diff
@@ -139,7 +159,7 @@ public SemanticDocumentCollector(SemanticSearchAttribute attribute, ISearchProvi
 
         public Task AddAsync(SearchableDocument item, CancellationToken cancellationToken = default)
         {
-            if (item.ConnectionInfo == null)
+            if (item.ConnectionInfo == null || item.ConnectionInfo.CollectionName == null || item.ConnectionInfo.ConnectionName == null)
```

Comment:
Looks like you're checking for a null ConnectionName twice in the same expression. Did you mean to check a different property?

Reply:
When we serialize `ConnectionInfo`, the object is still initialized with its properties being null even if `ConnectionInfo` is not passed in by the user. I had to add this condition to account for that use case.

Reply:
Not sure if my comment was understood. The current code is checking `item.ConnectionInfo.CollectionName == null` _twice_, which is obviously a mistake.

Reply:
ohhhh that makes a lot more sense! I'll update this!